smt/surrogate_models/krg_based.py

```diff
@@ -1082,18 +1090,23 @@ def grad_minus_reduced_likelihood_function(log10t):
                     )
 
                     for i in range(len(self.noise0)):
+                        v= i + len(self.theta0)
+                        v2=deepcopy(v)
                         noise_bounds = np.log10(noise_bounds)
                         constraints.append(
-                            lambda log10t: log10t[i + len(self.theta0)]
+                            lambda log10t: log10t[v2]
                             - noise_bounds[0]
                         )
                         constraints.append(
                             lambda log10t: noise_bounds[1]
-                            - log10t[i + len(self.theta0)]
+                            - log10t[v2]
                         )
                         bounds_hyp.append(noise_bounds)
-                try:
+                theta_limits = np.repeat(np.log10([theta_bounds]),repeats=len(theta0), axis=0)
+                sampling = LHS(xlimits=theta_limits,criterion="maximin",random_state=41)
+                theta_all_loops = sampling(self.options["n_multistart"])
```

Comment:
I would suggest to np.vstack the theta0 from the user and the theta0s from the multistart (LHS) and just loop over the result. It should allow to suppress the "minimize duplication" and the optimal_theta_res_2?